core/src/main/scala/org/scalatra/servlet/RichResponse.scala

```diff
@@ -18,7 +18,7 @@ case class RichResponse(res: HttpServletResponse) {
   def status: ResponseStatus = ResponseStatus(res.getStatus)
 
   def status_=(statusLine: ResponseStatus): Unit = {
-    res.setStatus(statusLine.code, statusLine.message)
+    res.sendError(statusLine.code, statusLine.message)
```

Comment:
`sendError()` produces the error page, so response body is not same as the case of `setStatus()`. You should update also test cases if you update this line to use `sendError()`.

Reply:
I understand.

Reply:
At least, in Jetty, `sendError()` proceeds the error page if status >= 200. This might be unexpected behavior in some cases. We might shouldn't use `sendError()` here. How about dropping `ResponseStatus`, and handling only status code?

Reply:
I thought about the intention that setStatus with two arguments became deprecated.

In the case of 4xx or 5xx, I understood that it is a policy to explain the reason by status_line or explain the reason in the body, and not to use both.

In other words, I understood that you need to use setStatus and sendError properly.

It is necessary to change API.

I will withdraw the PR once and I will redirect the issue.

Reply:
> I thought about the intention that setStatus with two arguments became deprecated.

Agree. I think that we should drop the reason parameter in the future. So we should mark current API as deprecated and offer the alternative API in 2.6.0.

Although, replacing `isRequestedSessionIdFromUrl` with `isRequestedSessionIdFromURL` is no problem. Could you create an another pull request for it? or you can push it directly because it's tiny fix and already reviewed in this pull request.

Reply:
Agree.

I will create another PR for `isRequestedSessionIdFromUrl`